core/templates/dev/head/editor/feedback_tab/ThreadDataService.js

```diff
@@ -50,7 +50,8 @@ oppia.factory('threadDataService', [
   };
 
   var _fetchMessages = function(threadId) {
-    $http.get(_THREAD_HANDLER_PREFIX + threadId).success(function(data) {
+    $http.get(_THREAD_HANDLER_PREFIX 
+        + threadId.split(".")[1]).success(function(data) {
```

Comment:
I'm starting to wonder, instead of doing all this joining and splitting, is it possible to just use the 'short' thread id throughout, and get rid of the concept of "exp id + thread id" until the very last moment it's needed (to put things into the storage layer)?